charon/tests/ui/issue-94-recursive-trait-defns.rs

```diff
@@ -1,4 +1,3 @@
-//@ known-panic
 pub trait Trait1 {
     type T: Trait2;
 }
```

Comment:
Could you add more examples of ways for traits to be mutually recursive? E.g. what you added to the other file, mutual reference via method signatures, via method calls, implicit reference due to parent traits, etc, as well as examples that could look recursive but aren't.

Reply:
I moved the tests to charon/tests/ui (as they are unsupported by Aeneas) and added a few extra tests.
As `dyn` is currently unsupported, I do not think I can add tests using method signatures, but please feel free to expand if you have other ideas of patterns to test.